### PR TITLE
Fix OnDiskMavenDownloader path not resolving properly on windows

### PR DESCRIPTION
**Pull request by janicduplessis**
OnDiskMavenDownloader#fetch used root.resolve to convert the uri returned from the MavenUrlDecoder to a local file scheme. This did not work on Windows because of the drive letter.
```
file:///C:/com/android/support/support-v4/23.0.1/support-v4-23.0.1.aar
```
on windows vs
```
file:///com/android/support/support-v4/23.0.1/support-v4-23.0.1.aar
```
on mac/linux

This removes the extra uri conversion that was breaking the path because it is only used to log the DownloadEvent which can use the actual target path instead.

Tested by downloading support-v4 while building react-native on both mac and windows.

Fixes the 2nd issue in #649

### CODE REVIEW

**Comment by janicduplessis** [comment]
cc @sdwilsh

**Comment by sdwilsh** [comment]
Could you also add a test for this?  You can create a new class, `OnDiskMavenDownloaderIntegrationTest`, and write a simple test that downloads from the SDK.  I want to make sure we don't regress this again or on Unix systems now.

**Comment by janicduplessis** [comment]
There's already `OnDiskMavenDownloaderTest` but it tests using a mock unix filesystem so it didn't catch this issue. I can add a test like `shouldDownloadFileFromLocalMavenRepo` that uses a windows mock filesystem instead or maybe run all the tests in this file on the 3 different file system mocks (unix, osx, windows).

**Comment by janicduplessis** [comment]
Does something like this looks good? This new test breaks before my fix.

File: test/com/facebook/buck/file/OnDiskMavenDownloaderTest.java [hunk 1]
```diff
@@ -84,6 +84,29 @@ public void shouldDownloadFileFromLocalMavenRepo() throws URISyntaxException, IO
   }
 
   @Test
+  public void shouldDownloadFileFromLocalMavenRepoOnWindows() throws URISyntaxException, IOException {
+    // Same test as `shouldDownloadFileFromLocalMavenRepo` but using a windows
+    // mock filesystem.
```
Comment (sdwilsh) [inline test/com/facebook/buck/file/OnDiskMavenDownloaderTest.java]: I wouldn't bother with this comment; the name implies this.

File: test/com/facebook/buck/file/OnDiskMavenDownloaderTest.java [hunk 2]
```diff
@@ -84,6 +84,29 @@ public void shouldDownloadFileFromLocalMavenRepo() throws URISyntaxException, IO
...
+    Path source = root.resolve("group/project/0.1/project-0.1.jar");
+    Files.createDirectories(source.getParent());
+    Files.write(source, "cake".getBytes(UTF_8));
+
+    Downloader downloader = new OnDiskMavenDownloader(root);
+    downloader.fetch(BuckEventBusFactory.newInstance(), uri, output);
+
+    String result = new String(Files.readAllBytes(output), UTF_8);
+
+    assertEquals("cake", result);
```
Comment (sdwilsh) [inline test/com/facebook/buck/file/OnDiskMavenDownloaderTest.java]: nit: use `assertThat(result, Matchers.equals("cake"));`

**Comment by sdwilsh** [comment]
That looks good; just fix the two comments please :)

**Comment by janicduplessis** [comment]
There you go!

**Comment by sdwilsh** [comment]
@facebook-github-bot shipit

**Comment by sdwilsh** [comment]
Whoops.  You've got some lint failures that you can see if you run `ant lint`.  CI is busted right now, but @bolinfest is looking into getting that resolved.

**Comment by janicduplessis** [comment]
Oupsi, fixed.

**Comment by sdwilsh** [comment]
@facebook-github-bot import

**Pull request closed without merging**